src/llvm-gcroot.cpp

```diff
@@ -37,6 +42,74 @@ static struct {
 } jl_gc_frame_stats = {0};
 #endif
 
+typedef std::pair<CallInst*, unsigned> frame_register;
+struct liveness {
+    typedef unsigned id;
+    enum {
+        // an assignment to a gcroot exists in the basic-block
+        // (potentially no live-in from the predecessor basic-blocks)
```

Comment:
I was just reading some doc about liveness analysis and I've seen a few use of it.